azext_iot/dps/providers/discovery.py

```diff
@@ -75,5 +75,12 @@ def _build_target(
         result["secondarykey"] = policy.secondary_key
         result["subscription"] = self.sub_id
         result["cmd"] = self.cmd
+        result["idscope"] = resource.properties.id_scope
 
         return result
+
+    def get_id_scope(self, resource_name: str) -> str:
```

Comment:
nit, theoretically could take an optional resource_group param even if its not used today.